youroom/static/js/publicacion.js

```diff
@@ -0,0 +1,155 @@
+class Etiqueta {
+    constructor(nombre, enlace, coordX, coordY) {
+        this.nombre = nombre;
+        this.enlace = enlace;
+        this.coordX = coordX;
+        this.coordY = coordY;
+    }
+
+    set addEnlace(enlace) {
+        this.enlace = enlace;
+    }
+}
+
+let contadorEtiqueta = 0;
+let etiqueta = null;
+let listaEtiquetas = [];
+
+function procesarEtiquetas(listaEtiquetas){
+    let etiquetasStr = "";
+    for(i = 0; i < listaEtiquetas.length; i++){
+        let etiqueta = listaEtiquetas[i];
+        etiquetasStr += etiqueta.enlace + "," + etiqueta.coordX + "," + etiqueta.coordY + "|"
+    }
+    return etiquetasStr.substring(0, etiquetasStr.length - 1);
+}
+
+function validaEnlace(enlace) {
+    let validador = false;
+    if(/^(http|https|ftp):\/\/?.*$/i.test(enlace)){
+        validador = true;
+    }
+
+    return validador;
+}
+
+function readURL(input) {
+    let validador = true;
+    if (input.files && input.files[0]) {
+        const tipo = input.files[0].type;
+        if (tipo == "image/jpg" || tipo == "image/jpeg" || tipo == "image/png") {
+            let reader = new FileReader();
+            reader.onload = function (e) {
+                $('#visor-imagen').attr('src', e.target.result);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/youRoom-ISPP/youRoom/pullRequest?prid=7157942)
```suggestion
                $("#visor-imagen").attr('src', e.target.result);
```